src/main/scala/devices/gpio/GPIO.scala

```diff
@@ -103,54 +111,143 @@ abstract class GPIO(busWidthBytes: Int, c: GPIOParams)(implicit p: Parameters)
   val rise = ~valueReg & inSyncReg;
   val fall = valueReg & ~inSyncReg;
 
-  val iofEnFields =  if (c.includeIOF) (Seq(RegField.rwReg(c.width, iofEnReg.io,
+  implicit val mode64: Boolean = c.width > 32
```

Comment:
Rather than perpetuating and making worse all this tech debt packing and unpacking into units, why not just represent each as a Seq of bools, 1-bit wide RegFields? Or use the built in “RegMap.bytes” function that already does this?

If the async registers are a problem, switch this file to use proper chisel3 AsyncReset